### ISSUE TITLE
re: git repo out of date possibly?

### ISSUE THREAD

**Issue by HBL001** [KyleLopin/PSoC-Potentiostat#17]
Hi @KyleLopin @Zanith 

I am building myself a Potentiostat circuit like the one in your paper, it's a really nice approach.  Plus I am going to see if we can squeeze a little more out of it by doing a few things including putting a little pre-amp, the aim to measure nanoamps ( I will share details when its done).  Anyway I have a different board to the type "CY8CKIT-059" its is a "CY8CKIT-50" that has the same processor but a little breakout area and a funky 2 line LCD display add-on.  I was given this by a friendly Cypress rep before they were taken over by Infineon, and its really the first time I have ever used it. 

The CY8CKIT-50B kit has the CY8C5868AXI-LP035 processor (  CY8C5868AXI-LP035 and  CY8C5888LTI-LP097 are essentially the same M3 but) with a different form factor.

- CY8C5868AXI-LP035: Comes in a 100-TQFP package.
- CY8C5888LTI-LP097: Comes in a 68-QFN package
 
  So the programmer just refuses to download the Amperometry_v059_2.hex file because the pin mapping/definitions are different, and 

![image](https://github.com/user-attachments/assets/c6525a66-09f6-484c-af8f-a99d876e3dc5)

This does mean I got a little work to change the pin mapping and recompile and flash the chip. 

So I am using the PSOC Creator 4.4 and cloned the PSOC repo and tried to open the workspace 

![image](https://github.com/user-attachments/assets/1fa5e943-8046-4490-9213-5dc2ca3d40fa)

At this point, I had a little look into the history of the codebase and notice considerable work to provision the UART and move away from the USB.   

Is it likely that these workspace/project files have not kept pace with the rest of the codebase?

![image](https://github.com/user-attachments/assets/d55a9e1b-caac-443c-864d-052d0d226e58)

I am a C/C++ programmer but not a PSOC expert, maybe the exercise of trying to figure this out will go something towards fixing that.  Some discussion to aid me or an update to this codebase would be brilliant as I would rather be doing electrochemistry !

Your help would be greatly appreciated.

Regards

Rich

**Comment by HBL001** [comment]
I went back in time to 2 years ago and checked out b83135e and for the moment I can have a try

**Comment by HBL001** [comment]
I got the other board programmed just fine, but the GUI now just refuses to acknowledge the board is connected.  Yes, I have swapped ports.

**Comment by KyleLopin** [comment]
You can't load the hex into the 050 kit as it is compiled for the 059 kit.  You will have to download all the code into PSoC Creator and compile it and upload with PSoC Creator. there are 2 USB ports, 1 from programming and 1 for using with GUI.  Also did you install USB library with libusb?

**Comment by HBL001** [comment]
Hi Kyle.  Yes this is why I am accessing the codebase to change the configuration and recompile it for the 050 kit.   It's the first time I have used the PSOC system and the error messages gave me a jump scare.  The error messages are just due to some unused files.  The latest commit is fine to use, ignore the errors and delete the items marked with the "no way" markers.  My fork has this cleared away, I also made a branch on my fork called CY8CKIT-50 which has the configuration changes and I have reimplemented the LCD 2 line display - the messages from the front-end is reassuring.    It all seems to work fine now.  I have fabricated a glucose test strip connector and everything is looking good for some lab work
